os/mm/mm_heap/mm_getheap.c

```diff
@@ -81,9 +81,19 @@ int mm_get_heapindex(void *mem)
 		return 0;
 	}
 	for (heap_idx = 0; heap_idx < CONFIG_MM_NHEAPS; heap_idx++) {
-		if (mem < (void *)(g_mmheap[heap_idx].mm_heapstart + g_mmheap[heap_idx].mm_heapsize) && mem >= (void *)g_mmheap[heap_idx].mm_heapstart) {
-			return heap_idx;
+		int region = 0;
+#if CONFIG_MM_REGIONS > 1
+		for (; region < g_mmheap[heap_idx].mm_nregions; region++)
+#endif
+		{
+			/* A valid address from the user heap for this region would have to lie
+			 * between the region's two guard nodes.
+			 */
+			if ((mem > (void *)g_mmheap[heap_idx].mm_heapstart[region]) && (mem < (void *)g_mmheap[heap_idx].mm_heapend[region])) {
```

Comment:
Isn't it should be >= and <= ?

Reply:
Thanks for your review!
```text
Region *:
A(addr)      B            C      D            E      F     ...      Y            Z
+------------+------------+------+------------+------+--------------+------------+
| guard node | alloc node + mem1 | alloc node + mem2 |     ...      | guard node |
+--------------------------------------------------------------------------------+
```
My understanding is, the input `mem` should be an pointer to the allocated memory, 
like `C->mem1` and `E->mem2` above. So that, it must be between the address  `B` and `Y`.
I think `> and <` can work fine. Of course, `>= and <=` is also correct.

Reply:
I think `> and <` is enough.